tests/integration/modules/mac_shadow.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+'''
+integration tests for mac_system
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import datetime
+import random
+import string
+
+# Import Salt Testing libs
+from salttesting.helpers import ensure_in_syspath, destructiveTest
+from salt.ext.six.moves import range
+ensure_in_syspath('../../')
+
+# Import salt libs
+import integration
+import salt.utils
+
+
+def __random_string(size=6):
+    '''
+    Generates a random username
+    '''
+    return 'RS-' + ''.join(
+        random.choice(string.ascii_uppercase + string.digits)
+        for x in range(size)
+    )
+
+
+TEST_USER = __random_string()
+
+
+class MacShadowModuleTest(integration.ModuleCase):
+    '''
+    Validate the mac_system module
+    '''
+
+    def setUp(self):
+        '''
+        Get current settings
+        '''
+        if not salt.utils.is_darwin():
+            self.skipTest('Test only available on Mac OS X')
+
+        if not salt.utils.which('dscl'):
+            self.skipTest('Test requires dscl binary')
+
+        if not salt.utils.which('pwpolicy'):
+            self.skipTest('Test requires pwpolicy binary')
+
+        if salt.utils.get_uid(salt.utils.get_user()) != 0:
+            self.skipTest('Test requires root')
+
+        super(MacShadowModuleTest, self).setUp()
+
+    def tearDown(self):
+        '''
+        Reset to original settings
+        '''
+        self.run_function('user.delete', [TEST_USER])
+
+        super(MacShadowModuleTest, self).tearDown()
+
+    @destructiveTest
+    def test_info(self):
+        '''
+        Test shadow.info
+        '''
+        self.run_function('user.add', [TEST_USER])
+        ret = self.run_function('shadow.info', [TEST_USER])
+        self.assertTrue(isinstance(ret, dict))
+        self.assertEqual(ret['name'], TEST_USER)
+
+    @destructiveTest
+    def test_get_account_created(self):
+        '''
+        Test shadow.get_account_created
+        '''
+        self.run_function('user.add', [TEST_USER])
+        text_date = self.run_function('shadow.get_account_created', [TEST_USER])
+        obj_date = datetime.datetime.strptime(text_date, '%Y-%m-%d %H:%M:%S')
+        self.assertTrue(isinstance(obj_date, datetime.date))
```

Comment:
This doesn't make sense to me. It seems like what you're doing here is effectively checking to see if `strptime` can convert the timestamp, but if it can't, you'll raise a `ValueError` (per the documentation) and you'll never even _reach_ this assert. This problem exists throughout this test class.